### PR TITLE
Master Update

### PR DESCRIPTION
**Pull request by Justman100**

### CODE REVIEW

**Comment by Justman100** [comment]
@MacFan4000

**Comment by Justman100** [comment]
@MacFan4000

File: LocalSettings.php [hunk 1]
```diff
@@ -201,199 +213,344 @@
...
 $wgGroupPermissions['user']['translate-groupreview'] = true;
 $wgGroupPermissions['user']['translate-import'] = true;
-$wgGroupPermissions['user']['createtalk'] = false;
-$wgGroupPermissions['user']['createpage'] = false;
-$wgGroupPermissions['user']['edit'] = false;
-$wgGroupPermissions['user']['read'] = false;
+$wgGroupPermissions['user']['translate-messagereview'] = true;
+$wgGroupPermissions['user']['user'] = true;
 $wgGroupPermissions['user']['writeapi'] = false;
-$wgGroupPermissions['bot']['writeapi'] = false;
```
Comment (MacFan4000) [inline LocalSettings.php:242]: This seems to be removed completely?

File: LocalSettings.php [hunk 2]
```diff
@@ -201,199 +213,344 @@
...
+$wgGroupPermissions['protecter']['editprotected'] = true;
+$wgGroupPermissions['protecter']['editsemiprotected'] = true;
+
+/// Deleter
+$wgGroupPermissions['deleter']['delete'] = true;
+$wgGroupPermissions['deleter']['bigdelete'] = true;
+$wgGroupPermissions['deleter']['deleterevision'] = true;
+$wgGroupPermissions['deleter']['deletechangetags'] = true;
+
+/// Abusefilter restricted users
```
Comment (MacFan4000) [inline LocalSettings.php:304]: this seems stripped down from what it currently is?

File: LocalSettings.php [hunk 3]
```diff
@@ -201,199 +213,344 @@
...
+$wgGroupPermissions['translateadmin']['pagetranslation'] = true;
+$wgGroupPermissions['translateadmin']['translate-manage'] = true;
+
+/// Chat moderators
+$wgGroupPermissions['chatmod']['chat'] = false;
+
+/// Interwiki administrators
+$wgGroupPermissions['interwiki-admin']['interwiki'] = true;
+
+/// Protecter
```
Comment (MacFan4000) [inline LocalSettings.php]: why are we adding protecter and deleter?

File: LocalSettings.php [hunk 4]
```diff
@@ -201,199 +213,344 @@
...
+	'autoconfirmed',
+	'bureaucrat',
+	'confirmed',
+	'patroller',
+	'steward',
+	'suppress',
+	'sysop',
+	'translateadmin'
+);
+$wgRemoveGroups['steward'] = array(
```
Comment (MacFan4000) [inline LocalSettings.php:484]: stewards are currently able to remove steward, that shouldn't change.

File: LocalSettings.php [hunk 5]
```diff
@@ -201,199 +213,344 @@
...
+	'autoconfirmed',
+	'bureaucrat',
+	'confirmed',
+	'patroller',
+	'suppress',
+	'sysop',
+	'translateadmin'
+);
+
+/// System administrators
```
Comment (MacFan4000) [inline LocalSettings.php]: this is redundant to sysadmins having the "userrights" permission.

File: LocalSettings.php [hunk 6]
```diff
@@ -201,199 +213,344 @@
...
+	'sysop',
+	'bureaucrat',
+	'steward',
+	'sysadmin'
+);
+
+$wgSemiprotectedRestrictionLevels = array(
+	'user',
+	'autoconfirmed',
+	'translateadmin'
```
Comment (MacFan4000) [inline LocalSettings.php]: "translateadmin" is not a valid restriction level.

File: LocalSettings.php [hunk 7]
```diff
@@ -201,199 +213,344 @@
...
+    unset( $wgGroupPermissions['staff']
+);
+    unset( $wgGroupPermissions['editor']
+);
+
 };
-$wgShowExceptionDetails = true;
 
-$wgGrantPermissions['editprotected']['bureaucrat'] = true;
+// RC fed settings
```
Comment (MacFan4000) [inline LocalSettings.php]:
```suggestion
// RC feed settings
```

File: LocalSettings.php [hunk 8]
```diff
@@ -201,199 +213,344 @@
...
+	'uri' =>
+	'udp://129.153.11.72:9872',
+	'formatter' => IRCColourfulRCFeedFormatter::class,
+	'add_interwiki_prefix' => false,
+	'omit_bots' => true,
+];
+
+// Other settings
 
-// Non-Steward suppressors
```
Comment (MacFan4000) [inline LocalSettings.php:393]: why is this group removed?

**Review by MacFan4000 — CHANGES_REQUESTED** [review]

File: LocalSettings.php (hunk 1 above)
Comment (Justman100) [reply LocalSettings.php:242]: Like mine? It's a bot! It MUST be able to write to the API! Is the purpose of a bot

File: LocalSettings.php (hunk 2 above)
Comment (Justman100) [reply LocalSettings.php:304]: Like mine?

File: LocalSettings.php (hunk 3 above)
Comment (Justman100) [reply LocalSettings.php]: For testing! For the completeness

File: LocalSettings.php (hunk 6 above)
Comment (Justman100) [reply LocalSettings.php]: Any role can be this?

File: LocalSettings.php (hunk 8 above)
Comment (Justman100) [reply LocalSettings.php:393]: Superfluous! One does not need a role that can simply give and take away a more powerful role! Then one simply assigns the more powerful role directly

Comment (Justarandomamerican) [reply LocalSettings.php:393]: That is against consensus. The consensus for NSSs was to allow certain people to add themselves suppressor temporarily using the non-stewardsuppress group, not assign it directly.

**Review by Justarandomamerican — CHANGES_REQUESTED** [review]
This is redundant because sysops can already grant and remove autopatrolled. I also thought crats could already remove crat?

File: LocalSettings.php (hunk 8 above)
Comment (Justman100) [reply LocalSettings.php:393]: What's the point of having a group only temporarily? Would it be better to apply for it permanently? And who is preventing them from having the group permanently?

File: LocalSettings.php (hunk 4 above)
Comment (Justman100) [reply LocalSettings.php:484]: See 997e304

**Comment by Justman100** [comment]
> This is redundant because sysops can already grant and remove autopatrolled.

?

> I also thought crats could already remove crat?

Yes, I originally wanted to suggest removing it too! But then I put it back again

**Comment by MacFan4000** [comment]
I undid most of the non cleanup changes, as things like group removal and certain rights changes require consensus.

**Pull request merged**